### ISSUE TITLE
AssertionError in WebSocketServer.removeConnection

### ISSUE THREAD

**Issue by greenrd** [TooTallNate/Java-WebSocket#512]
I am seeing this error occasionally:

```
java.lang.AssertionError: null
    at org.java_websocket.server.WebSocketServer.removeConnection(WebSocketServer.java:610)
    at org.java_websocket.server.WebSocketServer.onWebsocketClose(WebSocketServer.java:585)
    at org.java_websocket.WebSocketImpl.closeConnection(WebSocketImpl.java:492)
    at org.java_websocket.WebSocketImpl.closeConnection(WebSocketImpl.java:505)
    at org.java_websocket.WebSocketImpl.eot(WebSocketImpl.java:542)
    at org.java_websocket.SocketChannelIOHelper.read(SocketChannelIOHelper.java:17)
    at org.java_websocket.server.WebSocketServer.run(WebSocketServer.java:391)
...
```

which makes the server quit. From reading the code, it seems to me that this could be caused by a client closing the connection before the handshake has been sent.

**Comment by marci4** [comment]
Hello @greenrd,

thx for your bug report.
Could you provide some more additional infos?

- What version of the lib are you using?
- Are you using ws or wss?
- In what enviroment does this happen? OS? Java version?
- Do you know more about the client? 

Greetings
marci4

**Comment by marci4** [comment]
This issue should be resolved with pull request #514.

Closing the issue!
Greetings
marci4
